all: Add Standard service and characteristic UUIDs

deadprogram opened the pull request:
This PR adds generated definitions for the standard service and characteristic UUIDs from the https://github.com/NordicSemiconductor/bluetooth-numbers-database

deadprogram committed:
gen: generate standard service and characteristic UUIDs from Nordic Semiconductor bluetooth numbers database

Signed-off-by: deadprogram <ron@hybridgroup.com>

deadprogram committed:
examples: use standard service and characteristic UUIDs

Signed-off-by: deadprogram <ron@hybridgroup.com>

aykevl commented on Makefile:
```diff
@@ -46,3 +47,9 @@ smoketest-macos:
+gen-uuids:
+	# generate the standard service and characteristic UUIDs
+	go run ./bin/gen-service-uuids/main.go
+	go run ./bin/gen-characteristic-uuids/main.go
```
Nitpick: I think I would have placed this in some other directory than `bin` (for example, `tools`) because these aren't really binaries. But I'm fine with them as-is.

deadprogram replied:
Moved to `tools`

aykevl reviewed (commented):
I think it would be a good idea to include the license file in the data directory. Even though the license text is more or less the same, the JSON files are copyright by Nordic Semiconductors and I don't think we should strip that (especially when copying the files as-is): https://github.com/NordicSemiconductor/bluetooth-numbers-database/blob/master/LICENSE

Additionally, I'm a little bit worried about the format. How do you intend to use the name/identifier/source fields? Right now it would be hard to look them up, you'd have to check every global variable individually (no map lookup for example). On the other hand, putting them all in a big map would make it very hard to optimize all the unnecessary fields away. Perhaps only listing the UUIDs (no name/identifier/source) would make more sense?

deadprogram commented:
Updated license info, and also simplified the generated UUID to remove the extra info since as you point out there is really no way to use it at the moment.

I think this is now ready to go.

aykevl reviewed (approved):
Thank you, looks good to me!

aykevl commented:
I'm not sure whether to squash this or not? The first two commits are a sensible distinction but later commits are improvements to those first two.

deadprogram commented:
@aykevl I just squashed into 2 commits to keep things organized. Now ready for merge.
